src/util.cpp

```diff
@@ -16,7 +16,10 @@
 
 // clang-format off
 // Dont't remove `format off`, it prevent reordering of win-includes.
-#define _POSIX_C_SOURCE 200112L // For stat from stat/stat.h and fseeko() (POSIX extensions).
+
+#  define _XOPEN_SOURCE 700L
```

Comment:
These defines cause issues in base.h leading to failing android builds.

Reply:
Excuse me for stacking up so much builds, I was expecting old commits to get cancelled. I don't have a android sdk on my pc (and any experience with it) so this seemed easier.

Reply:
You can define global `FLATBUFFERS_LOCALE_INDEPENDENT=0` in your project to disable locale-independent code.
What goal of current changes?
How local-independent-code related to the title `Removes posix definition`?

Reply:
The goal is to have compliant code. For me PATH_MAX wasn't working on a POSIX conforming platform, have a look at one of the previous comments. So I switched to the more modern real_path which requires posix 2008 and as such XOPEN_SOURCE 700 (500 & 600 are for older POSIX).

Defining these leads to the activation of FLATBUFFERS_LOCALE_INDEPENDENT while this is actually not supported on a platform. I'm trying to fix that by using the feature test macros in a different manner.

I'll update the title.